name: Django CI

on:
  push:
    branches: [ "dev" ]
  pull_request:
    branches: [ "dev" ]

jobs:
  build:

    runs-on: ubuntu-latest
    strategy:
      max-parallel: 4
      matrix:
        python-version: [3.8, 3.9]

    steps:
    - uses: actions/checkout@v3
    - name: Set up Python ${{ matrix.python-version }}
      uses: actions/setup-python@v3
      with:
        python-version: ${{ matrix.python-version }}
    - name: Install Dependencies
      run: |
        python -m pip install --upgrade pip
        pip install -r requirements.txt
    - name: Run Tests
      env: 
        SECRET_KEY: 'django-insecure-41bn!!#i3!(_8q%%00wgs!4jmk(^w5!=+!kz+@qx&+d17@@y5!'
        DEBUG: False
        DB_NAME: coyete
        DB_USER: auth_wiki
        DB_PASSWORD: mypassword
        DB_HOST: 127.0.0.1
      run: |
       cd auth_wiki
        python manage.py test
